pkg/controller/controller.go

```diff
@@ -111,6 +111,18 @@ func (c *controller) serviceBindingCallback(e k8swatch.Event) error {
 			return err
 		}
 		glog.Infof("Created Service Binding: %s\n%v\n", sb.Name, created)
+	} else if e.Type == k8swatch.Deleted {
+		// if we've already processed the deletion event, ignore
```

Comment:
@MHBauer when the controller first intercepts a deleted event, it sets the deletion timestamp and re-creates the TPR (this is a half-measure that we implemented for now, to address the fact that we don't have an API server to intercept the API call for deletions. @bmelville and I know that almost everybody hates this. know that it will be fixed soon)

after the deletion timestamp is set and the TPR is re-created, the uninject, unbind, etc... actions are taken. when all of those actions are finished, the controller deletes the TPR again, but this time it has a deletion timestamp set on it. the deletion event is intercepted again, but since the deletion timestamp is set on the TPR, the event is ignored.

@bmelville wrote this very quickly, so it's very possible we missed a race somewhere. thoughts?